aten/src/ATen/Context.h

```diff
@@ -231,11 +224,16 @@ static inline bool hasMAGMA() {
 }
 
 static inline void manual_seed(uint64_t seed) {
-  globalContext().defaultGenerator(DeviceType::CPU).manualSeed(seed);
+  detail::getDefaultCPUGenerator()->setCurrentSeed(seed);
   // NB: Sometimes we build with CUDA, but we don't have any GPUs
   // available. In that case, we must not seed CUDA; it will fail!
   if (hasCUDA() && detail::getCUDAHooks().getNumGPUs() > 0) {
-    globalContext().defaultGenerator(DeviceType::CUDA).manualSeedAll(seed);
+    globalContext().lazyInitCUDA();
+    auto& generator = globalContext().generator_registry[static_cast<int>(DeviceType::CUDA)];
+    if(!generator) {
+      AT_ERROR(DeviceTypeName(DeviceType::CUDA), " backend type not enabled.");
```

Comment:
I think it's probably safe to say that this should never happen, if `lazyInitCUDA` worked OK. There are serious problems if this generator is not initialized at this point. Switch it to be an `AT_ASSERT`.